core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -455,22 +455,9 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
         producer.flush()
         assertTrue("All request are complete.", responses.forall(_.isDone()))
         // Check the messages received by broker.
-        val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
-        val maxIters = 50
-        var iters = 0
-        consumer1.seek(new TopicPartition(topic, 0),0)
```

Comment:
@baluchicken, thanks for the explanation. I think the seek is fine then. The error message is a bit more informative if we consume all of the data instead of just what was produced in the last iteration.